src/lib/wallet/GoodWalletClass.js

```diff
@@ -499,104 +486,116 @@ export class GoodWallet {
   }
 
   async pollSendEvents(toBlock, from = null) {
```

Comment:
not even reviewing all the rest. please revert all refactoring.

Reply:
in the next 3 methods I added just contract existence check